storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -128,56 +308,121 @@ private void updateSupervisorsResources(Cluster cluster, Topologies topologies)
-                str.append("\t-->Topology: " + topo_sched.getKey() + "\n");
-                for (Map.Entry<WorkerSlot, Collection<ExecutorDetails>> ws : topo_sched.getValue().entrySet()) {
-                    str.append("\t\t->Slot [" + ws.getKey().getPort() + "] -> " + ws.getValue() + "\n");
+        }
+
+        Map fromFile = Utils.findAndReadConfigFile("user-resource-pools.yaml", false);
+        Map<String, Map<String, Number>> tmp = (Map<String, Map<String, Number>>) fromFile.get(Config.RESOURCE_AWARE_SCHEDULER_USER_POOLS);
+        if (tmp != null) {
+            for (Map.Entry<String, Map<String, Number>> UserPoolEntry : tmp.entrySet()) {
+                String user = UserPoolEntry.getKey();
+                ret.put(user, new HashMap<String, Double>());
+                for (Map.Entry<String, Number> resourceEntry : UserPoolEntry.getValue().entrySet()) {
+                    ret.get(user).put(resourceEntry.getKey(), resourceEntry.getValue().doubleValue());
                 }
             }
         }
-        return str.toString();
+        return ret;
+    }
+
+    private SchedulingState checkpointSchedulingState() {
+        LOG.debug("/*********Checkpoint scheduling state************/");
+        for (User user : this.getUserMap().values()) {
+            LOG.debug(user.getDetailedInfo());
+        }
+        LOG.debug(ResourceUtils.printScheduling(this.cluster, this.topologies));
+        LOG.debug("nodes:\n{}", this.nodes);
+        LOG.debug("/*********End************/");
+        return new SchedulingState(this.userMap, this.cluster, this.topologies, this.nodes, this.conf);
+    }
+
+    private void restoreCheckpointSchedulingState(SchedulingState schedulingState) {
+        LOG.debug("/*********restoring scheduling state************/");
+        //reseting cluster
+        //Cannot simply set this.cluster=schedulingState.cluster since clojure is immutable
+        this.cluster.setAssignments(schedulingState.cluster.getAssignments());
+        this.cluster.setSupervisorsResourcesMap(schedulingState.cluster.getSupervisorsResourcesMap());
+        this.cluster.setStatusMap(schedulingState.cluster.getStatusMap());
+        this.cluster.setResourcesMap(schedulingState.cluster.getResourcesMap());
+        //don't need to explicitly set data structues like Cluster since nothing can really be changed
+        //unless this.topologies is set to another object
+        this.topologies = schedulingState.topologies;
+        this.conf = schedulingState.conf;
+        this.userMap = schedulingState.userMap;
+        this.nodes = schedulingState.nodes;
+
+        for (User user : this.getUserMap().values()) {
+            LOG.debug(user.getDetailedInfo());
+        }
+        LOG.debug(ResourceUtils.printScheduling(cluster, topologies));
+        LOG.debug("nodes:\n{}", this.nodes);
+        LOG.debug("/*********End************/");
```

Comment:
Could we just make a SchedulingState a member of the ResourceAwareScheduler?  Then restoring would be simply pointing to the desired SchedulingState instance.